compiler/runtime/OMRRuntimeAssumptions.hpp

```diff
@@ -185,6 +187,59 @@ class PatchNOPedGuardSite : public OMR::LocationRedirectRuntimeAssumption
    uint8_t *_destination;
    }; // TR::PatchNOPedGuardSite
 
+class PatchSites
+   {
+   private:
+   int       _refCount;
+   int       _size; 
+   uint8_t **_patchPoints;
+
+   public:
+   PatchSites(TR_PersistentMemory *pm, int size);
+
+   int getSize() { return _size; }
+   uint8_t *getLocation(size_t index) { return _patchPoints[index * 2]; }
+   uint8_t *getDestination(size_t index) { return _patchPoints[index * 2 + 2]; }
+
+   void set(size_t index, uint8_t *location, uint8_t *destination);
+   bool equals(PatchSites *other);
+   void addReference();
+   static void reclaim(PatchSites *sites);
+   };
+
+class PatchMultipleNOPedGuardSites : public OMR::LocationRedirectRuntimeAssumption
+   {
+   protected:
+   PatchMultipleNOPedGuardSites(TR_PersistentMemory *pm, uintptrj_t key, TR_RuntimeAssumptionKind kind, PatchSites *sites)
+      : OMR::LocationRedirectRuntimeAssumption(pm, key), _patchSites(sites) {}
+   public:
+
+   virtual void compensate(TR_FrontEnd *vm, bool isSMP, void *)
+      {
+      for (size_t i = 0; i < _patchSites->getSize(); ++i)
+         TR::PatchNOPedGuardSite::compensate(isSMP, _patchSites->getLocation(i), _patchSites->getDestination(i));
+      }
+
+   virtual bool equals(OMR::RuntimeAssumption &other)
+      {
+      PatchMultipleNOPedGuardSites *site = other.asPMNGSite(); 
+      return site != 0 && _patchSites->equals(site->getPatchSites());
+      }
+
+   virtual void reclaim()
+      {
+      PatchSites::reclaim(_patchSites);
+      }
+
+   virtual PatchMultipleNOPedGuardSites *asPMNGSite() { return this; }
+   virtual uint8_t *getAssumingPC() { return _patchSites->getLocation(0); }
```

Comment:
A comment on why index 0 is the right thing to return here would be helpful.

Reply:
I have updated the function with a comment. Existing uses of `getAssumingPC` were used to test whether the assumption was in an address range, such as a method. As all assumptions where assumed to be in the same method, retuning the first location appeared to be safe. As this restriction isn't explicitly enforced anywhere and comparisons may desire smaller ranges, `OMR::RuntimeAssumption` has been modified to consider a range of assuming PCs.